src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -123,7 +123,7 @@ public virtual TagBuilder GenerateAntiForgery([NotNull] ViewContext viewContext)
                 metadata,
                 name,
                 value: "true",
-                useViewData: !explicitValue,
+                useViewData: (metadata == null && !explicitValue),
```

Comment:
carried over from MVC 5.2 code which does the same `Remove` rather than passing a real `isExplicitValue` to it's equivalent of `GenerateInput`.  I'm not going to mess w/ this now because it's unrelated.  but I'll file a Task issue shortly...